src/Polly.Core/Hedging/OnHedgingArguments.cs

```diff
@@ -14,9 +14,14 @@ namespace Polly.Hedging;
     /// <summary>
     /// Initializes a new instance of the <see cref="OnHedgingArguments{TResult}"/> struct.
     /// </summary>
-    /// <param name="primaryContext">The outcome of the resilience operation or event.</param>
-    /// <param name="actionContext">The action context.</param>
+    /// <param name="primaryContext">The primary context received by hedging strategy.</param>
+    /// <param name="actionContext">The action context. cloned from the primary context.</param>
     /// <param name="attemptNumber">The zero-based hedging attempt number.</param>
+    /// <remarks>
+    /// The <paramref name="primaryContext"/> represents the context that was received by the hedging strategy and used to execute the primary action.
+    /// To prevent race conditions, hedging strategy then clones the primary context into <paramref name="actionContext"/> and uses it to execute the hedged action.
+    /// Every hedged action gets its own context that is cloned from primary.
```

Comment:
```suggestion
    /// To prevent race conditions, the hedging strategy then clones the primary context into <paramref name="actionContext"/> and uses it to execute the hedged action.
    /// Every hedged action gets its own context that is cloned from the primary.
```